codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ServerDowngrader.java

```diff
@@ -78,9 +80,12 @@ protected void decode(ChannelHandlerContext ctx, Http2StreamFrame frame, List<Ob
                     out.add(full);
                 }
             } else {
-                out.add(HttpConversionUtil.toHttpRequest(id, headersFrame.headers(), validateHeaders));
+                HttpRequest req = HttpConversionUtil.toHttpRequest(id, headersFrame.headers(), validateHeaders);
+                if (!HttpUtil.isContentLengthSet(req)) {
+                    HttpUtil.setTransferEncodingChunked(req, true);
```

Comment:
Oh, I see what you mean.  I thought you meant I could skip checking `if (!HttpUtil.isContentLengthSet(req)) {`.  I didn't realize you meant to skip using the `HttpUtil` method.

Sure, I'll change it tomorrow morning.